[3.8.] Avatar documentation (#1233)
telerik/blazor-docs: components/avatar/overview.md

---
title: Overview
page_title: Avatar Overview
description: Discover the Blazor Avatar and explore the examples.
slug: avatar-overview
tags: telerik,blazor,avatar,overview
published: True
position: 0
---

# Blazor Avatar Overview

The <a href="https://www.telerik.com/blazor-ui/avatar" target="_blank">Blazor Avatar component</a> is typically used to display images, icons or initials representing people or other entities.

You can set different [Avatar types]({%slug avatar-types%}) and customize its [size]({%slug avatar-appearance%}#size), [fill mode]({%slug avatar-appearance%}#fillmode) and more.

## Creating Blazor Avatar

1. Use the `TelerikAvatar` tag to add the component to your razor page.

> 1. Declare your desired content (as `ChildContent`) inside the `TelerikAvatar` tag - text, icon or image. This example demonstrates Avatar with initials(text).

Reviewer: ```suggestion
1. Declare your desired content (as `ChildContent`) inside the `TelerikAvatar` tag - text, icon or image. This example demonstrates Avatar with initials (text).
```

> 1. Set the [Avatar type]({%slug avatar-types%}) to `Text`.

Reviewer: ```suggestion
1. Set the [Avatar type]({%slug avatar-types%}). In this case, we are using `Text`.
```

---
title: Overview
page_title: Avatar Overview
description: Discover the Blazor Avatar and explore the examples.
slug: avatar-overview
tags: telerik,blazor,avatar,overview
published: True
position: 0
---

# Blazor Avatar Overview

The <a href="https://www.telerik.com/blazor-ui/avatar" target="_blank">Blazor Avatar component</a> is typically used to display images, icons or initials representing people or other entities.

It also gives you the flexibility to customize its size, color, fill mode and more.

> ## Creating Avatar

Reviewer: SEO

```suggestion
## Creating Blazor Avatar
```

1. Use the `TelerikAvatar` tag to add the component to your razor page.

> 1. Declare your desired content inside the `TelerikAvatar` tag - text, icon or image.

Reviewer 2: We can add the actual technical term for the space between the opening and closing tag - `ChildContent`.

Reviewer: Not sure how recognized is `ChildContent` by devs, but I agree that I stumbled upon this step and my first thought was "how"? Probably I expected a property and didn't understand that we are suggesting nested content inside the component tag.

Author: Can you please clarify to make sure we are on teh same page? Your suggestion is

```suggestion
1. Declare your desired `ChildContent` inside the `TelerikAvatar` tag - text, icon or image.
```

> 1. Set the [Avatar type]({%slug avatar-types%}) based on your content.

Reviewer: This step is not actionable unless the reader refers to other docs content. In such cases, we should either add one more sentence for clarification, or just suggest a specific setting for now and leave the other options for later.

1. (Optional) Configure the [`ThemeColor`]({%slug avatar-appearance%}) of the Avatar.

>caption Basic Avatar with text and specified `ThemeColor`

````CSHTML
@*Basic Avatar configuration*@

<TelerikAvatar Type="AvatarType.Text"
               ThemeColor="@Telerik.Blazor.ThemeConstants.Avatar.ThemeColor.Info">
    JD
</TelerikAvatar>
````

## Types

The Avatar `Type` defines the type of the content that the component will display - it may be text, icon or image. [Read more about the specifics of the Avatar types...]({%slug avatar-types%})

## Appearance

The Avatar provides multiple options for configuring its look. [Explore the available appearance settings...]({%slug avatar-appearance%})

> ## Parameters

Reviewer: ```suggestion
## Avatar Parameters
```

The Blazor Avatar provides the following parameters that allow you to configure the component:

> @[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

Reviewer 2: This template is no longer needed. Dimo made improvements to automatically add these styles to the page.

| Parameter    | Type  | Description |
| ----------- | ----------- | -------|
| `Bordered` | `bool` | Whether the Avatar will have border. |
| `Type` | `AvatarType` <br/> (`Image`) | The type of the Avatar. Read more in the [Types article]({%slug avatar-types%}). |

> ### Styling and Appearance

Reviewer: The number of parameters is small, so why split them into groups? Moreover, `Bordered` is for appearance as well, but it's in the "main" group.

Author: I wanted to keep the structure for dividing the parameters as in other components (e.g. [AutoComplete](https://docs.telerik.com/blazor-ui/components/autocomplete/overview#parameters)). However, here the number is indeed pretty small, so I will combine the tables.

The following parameters enable you to customize the appearance of the Blazor Avatar:

| Parameter | Type | Description |
| --- | --- | --- |

> | `Class` | `string` | The CSS class that will be rendered on the main wrapping element of the Avatar. (`<div class="k-avatar>`) Use it to [override the theme or apply custom styles]({%slug themes-override%}). |

Reviewer 2: | `Class` | `string` | Add a custom CSS class to the  `<div class="k-avatar>`. Use it to [override the theme or apply custom styles]({%slug themes-override%}). |

Author: This suggestion contradicts with the practice we've incorporated for the descriptions. The description should start with a noun and not a verb.

Reviewer 2: Ok, so how about - `A custom CSS class to the .....`?
